src/main/java/net/sf/mzmine/modules/masslistmethods/ADAPchromatogrambuilder/ADAPChromatogramBuilderTask.java

```diff
@@ -266,19 +261,18 @@ public void run() {
     // Stopwatch stopwatch3 = Stopwatch.createUnstarted();
 
 
-    processedPoints = 0;
-    totalPoints = simpleAllMzVals.length;
+    progress = 0.0;
+    double progressStep = 0.5 / simpleAllMzVals.length;

```

Comment:
any chance simpleAllMzVals.length might be 0 at this point?

Reply:
I guess it's possible. I wouldn't expect it to cause any errors since `progressStep` would never be assigned to `progress` in this case. But if you want I can set `progressStep` explicitly to 0.0 for example.

Reply:
This line will throw an ArithmeticException in case simpleAllMzVals.length == 0.
Same with ranges.size() below.
You should always explicitly check for zero when doing a division by an integer.